QuestionAnswering/QA_SEQ2SEQ/data_generation/locate_then_ask/singapore/locate.py

```diff
@@ -0,0 +1,130 @@
+        )
+        query_graph.add_triples(
+            [
+                ("Plot", PLOT_ATTR_2_PRED[key], measurevalue_node, dict(key=key)),
+                (measurevalue_node, "om:hasNumericalValue", numval_node),
+                (measurevalue_node, "om:hasUnit", unit_node),
+            ]
+        )
+
+        verbn = "whose {key} is {op} {unit}".format(
+            key=random.choice(PLOT_ATTR_LABELS[key]), op=op_verbn, unit=unit_verbn
+        )
+
+        return verbn
+
+    def locate_concept_and_literal_multi(self, entity_iri: str, cond_num: int):
+        query_graph, concept = self.locate_concept_name(entity_iri)
+
+        entity = self.store.get(entity_iri)
+        keys = entity.get_nonnone_keys()
+        weights = [self.ATTR_KEY_WEIGHTS[k] for k in keys]
+
+        conds = []        
+        for k in np.random.choice(
+            keys, size=min(cond_num, len(keys)), p=normalize_1d(weights), replace=False
+        ):
+            if k is OPltPlotAttrKey.LAND_USE_TYPE_TYPE:
+                assert entity.land_use_type_type is not None
+                assert entity.land_use_type_type.startswith(OZNG)
+                clsname = entity.land_use_type_type[len(OZNG) :]
+                clsname_node = "{prefix}:{name}".format(
+                    prefix=NAMESPACE2PREFIX[OZNG], name=clsname
+                )
+                query_graph.add_iri_node(clsname_node, prefixed=True)
+                query_graph.add_triple("Plot", PLOT_ATTR_2_PRED[k], clsname_node, key=k)
+                cond = "whose {attr} is {value}".format(
+                    attr=random.choice(["land use type", "land use classification"]),
+                    value=random.choice(OZNG_LANDUSETYPE_LABELS[clsname]),
+                )
+            elif k is OPltPlotAttrKey.GROSS_PLOT_RATIO:
+                assert entity.gross_plot_ratio is not None
+                cond = self._locate_measure(
+                    query_graph, key=k, measure=entity.gross_plot_ratio
+                )
+            elif k is OPltPlotAttrKey.IS_AWAITING_DETAILED_GPR_EVAL:
+                assert entity.is_awaiting_detailed_gpr_eval is not None
+                literal_node = query_graph.make_literal_node(
+                    entity.is_awaiting_detailed_gpr_eval
+                )
+                query_graph.add_triple("Plot", PLOT_ATTR_2_PRED[k], literal_node, key=k)
+                if entity.is_awaiting_detailed_gpr_eval:
+                    cond = "which is awaiting detailed gross plot ratio evaluation"
+                else:
+                    cond = "which is not awaiting detailed gross plot ratio evaluation"
+            elif k is OPltPlotAttrKey.PLOT_AREA:
+                assert entity.plot_area is not None
+                cond = self._locate_measure(
+                    query_graph, key=k, measure=entity.plot_area
+                )
+            elif k is OPltPlotAttrKey.GROSS_FLOOR_AREA:
+                assert entity.gross_floor_area is not None
```

Comment:
See above comment. Several occurrences in this file.

Reply:
I've replaced the `assert` statement with `raise ValueError(...)`.